src/state.rs

```diff
@@ -262,11 +303,13 @@ impl Default for BlendChannel {
 }
 
 #[allow(missing_docs)]
-#[derive(Copy, Clone, PartialOrd, PartialEq)]
+#[derive(Copy, Clone, Hash, PartialOrd, PartialEq, Eq)]
 pub struct Blend {
     pub color: BlendChannel,
     pub alpha: BlendChannel,
-    pub value: target::ColorValue,
+    /// Color mask to use. Each flag indicates that the given color channel can be written to, and
+    /// they can be OR'd together.
+    pub mask: ColorMask,
```

Comment:
This is incorrect. We should be able to specify color mask without any blending. I suggest:
```rust
pub struct Color {
     pub mask: ColorMask,
     pub blend: Option<Blend>,
}
```
This would also allow adding more per-target states, if needed (GL supports `MULTISAMPLE` and `DITHER` states per target, but we don't need to rush it until we compare with other APIs).